torchmultimodal/modules/encoders/albef_text_encoder.py

```diff
@@ -0,0 +1,324 @@
+            extended_attention_mask = attention_mask[:, None, :, :]
+        elif attention_mask.dim() == 2:
+            extended_attention_mask = attention_mask[:, None, None, :]
+        else:
+            raise ValueError(
+                "Wrong shape for attention_mask (shape {})".format(attention_mask.shape)
+            )
+
+        # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
+        # masked positions, this operation will create a tensor which is 0.0 for
+        # positions we want to attend and -10000.0 for masked positions.
+        # Since we are adding it to the raw scores before the softmax, this is
+        # effectively the same as removing these entirely.
+        extended_attention_mask = extended_attention_mask.to(
+            dtype=attention_mask.dtype
+        )  # fp16 compatibility
+        extended_attention_mask = (1.0 - extended_attention_mask) * -10000.0
+        return extended_attention_mask
+
+    def forward(
+        self,
+        input_ids: Tensor = None,
+        attention_mask: Optional[Tensor] = None,
+    ) -> Tensor:
+        input_shape = input_ids.size()
+        device = input_ids.device
+        if attention_mask is None:
+            attention_mask = torch.ones(input_shape, device=device)
+
+        extended_attention_mask = self.get_extended_attention_mask(attention_mask)
+
+        embedding_output = self.embeddings(input_ids)
+        encoder_outputs = self.encoder(embedding_output, extended_attention_mask)
+
+        return encoder_outputs
+
+
+class ALBEFTextEmbeddings(nn.Module):
+    def __init__(
+        self,
+        vocab_size: int,
+        hidden_size: int,
+        padding_idx: int,
+        max_position_embeddings: int,
+        type_vocab_size: int,
+        layer_norm_eps: float,
+    ) -> None:
+        super().__init__()
+        self.word_embeddings = nn.Embedding(vocab_size, hidden_size, padding_idx)
+        self.position_embeddings = nn.Embedding(max_position_embeddings, hidden_size)
+        self.token_type_embeddings = nn.Embedding(type_vocab_size, hidden_size)
+        self.LayerNorm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+
+    def forward(self, input_ids: Tensor) -> Tensor:
+        input_shape = input_ids.size()
+        seq_length = input_shape[1]
+        device = input_ids.device
+
+        position_ids = torch.arange(seq_length, dtype=torch.long, device=device)
+        position_ids = position_ids.unsqueeze(0).expand(input_shape)
+        token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=device)
+
+        inputs_embeds = self.word_embeddings(input_ids)
+        position_embeddings = self.position_embeddings(position_ids)
+        token_type_embeddings = self.token_type_embeddings(token_type_ids)
+
+        embeddings = inputs_embeds + position_embeddings + token_type_embeddings
+        embeddings = self.LayerNorm(embeddings)
+        return embeddings
+
+
+class ALBEFEncoder(nn.Module):
```

Comment:
Can we name this something else? `ALBEFEncoder` and `ALBEFTextEncoder` in the same file (not to mention other `ALBEF*Encoders` we have floating around) can be confusing